diff --git a/third_party/emdawnwebgpu/webgpu.cpp b/third_party/emdawnwebgpu/webgpu.cpp
index 5bfac41dcc..71a153daaa 100644
--- a/third_party/emdawnwebgpu/webgpu.cpp
+++ b/third_party/emdawnwebgpu/webgpu.cpp
@@ -692,6 +692,7 @@ struct WGPUBufferImpl final : public EventSource,
   WGPUBufferImpl(const EventSource* source, bool mappedAtCreation);
   // Injection constructor used when we already have a backing Buffer.
   WGPUBufferImpl(const EventSource* source, WGPUBufferMapState mapState);
+  ~WGPUBufferImpl();
 
   void Destroy();
   const void* GetConstMappedRange(size_t offset, size_t size);
@@ -1361,6 +1362,12 @@ WGPUBufferImpl::WGPUBufferImpl(const EventSource* source,
       RefCountedWithExternalCount(kImportedFromJS),
       mMapState(mapState) {}
 
+WGPUBufferImpl::~WGPUBufferImpl() {
+  if (!IsImported()) {
+    Destroy();
+  }
+}
+
 void WGPUBufferImpl::Destroy() {
   emwgpuBufferDestroy(this);
   AbortPendingMap("Buffer was destroyed before mapping was resolved.");
diff --git a/src/tint/utils/memory/aligned_storage.h b/src/tint/utils/memory/aligned_storage.h
index c532c4fc38..19c950af4c 100644
--- a/src/tint/utils/memory/aligned_storage.h
+++ b/src/tint/utils/memory/aligned_storage.h
@@ -31,6 +31,9 @@
 #include <cstddef>
 
 #include "src/tint/utils/memory/bitcast.h"
+#include "src/tint/utils/macros/compiler.h"
+
+TINT_BEGIN_DISABLE_WARNING(UNSAFE_BUFFER_USAGE);
 
 namespace tint {
 
@@ -50,4 +53,6 @@ struct alignas(alignof(T)) AlignedStorage {
 
 }  // namespace tint
 
+TINT_END_DISABLE_WARNING(UNSAFE_BUFFER_USAGE);
+
 #endif  // SRC_TINT_UTILS_MEMORY_ALIGNED_STORAGE_H_
