diff --git a/include/onnxruntime/core/graph/graph.h b/include/onnxruntime/core/graph/graph.h
index 81015b2..19caa69 100644
--- a/include/onnxruntime/core/graph/graph.h
+++ b/include/onnxruntime/core/graph/graph.h
@@ -20,6 +20,8 @@
 #pragma warning(pop)
 #endif
 
+#include "flatbuffers/flatbuffers.h"
+
 #include "core/common/gsl.h"
 
 #include "core/common/common.h"
@@ -43,12 +45,6 @@
 #include "core/graph/node_arg.h"
 #include "core/graph/ort_format_load_options.h"
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-template <typename T>
-struct Offset;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 class Graph;
 struct IndexedSubGraph;
diff --git a/onnxruntime/core/flatbuffers/flatbuffers_utils.h b/onnxruntime/core/flatbuffers/flatbuffers_utils.h
index 4e7db4d..55bde0b 100644
--- a/onnxruntime/core/flatbuffers/flatbuffers_utils.h
+++ b/onnxruntime/core/flatbuffers/flatbuffers_utils.h
@@ -5,6 +5,8 @@
 
 #include <unordered_map>
 
+#include "flatbuffers/flatbuffers.h"
+
 #include "core/common/common.h"
 #include "core/common/path_string.h"
 #include "core/common/status.h"
@@ -13,18 +15,6 @@ namespace ONNX_NAMESPACE {
 class ValueInfoProto;
 }
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-
-template <typename T>
-struct Offset;
-
-struct String;
-
-template <typename T>
-class Vector;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 
 namespace fbs {
diff --git a/onnxruntime/core/framework/kernel_type_str_resolver.h b/onnxruntime/core/framework/kernel_type_str_resolver.h
index 75fc2fa..31a806d 100644
--- a/onnxruntime/core/framework/kernel_type_str_resolver.h
+++ b/onnxruntime/core/framework/kernel_type_str_resolver.h
@@ -7,6 +7,8 @@
 #include <string_view>
 #include <utility>
 
+#include "flatbuffers/flatbuffers.h"
+
 #if !defined(ORT_MINIMAL_BUILD)
 #include "core/graph/onnx_protobuf.h"
 #endif  // !defined(ORT_MINIMAL_BUILD)
@@ -18,12 +20,6 @@
 #include "core/graph/graph.h"
 #include "core/platform/ort_mutex.h"
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-template <typename T>
-struct Offset;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 
 namespace fbs {
diff --git a/onnxruntime/core/framework/session_state.h b/onnxruntime/core/framework/session_state.h
index d546f26..51bb029 100644
--- a/onnxruntime/core/framework/session_state.h
+++ b/onnxruntime/core/framework/session_state.h
@@ -8,6 +8,8 @@
 #include <unordered_map>
 #include <vector>
 
+#include "flatbuffers/flatbuffers.h"
+
 #include "core/common/gsl.h"
 
 #include "core/common/common.h"
@@ -43,12 +45,6 @@
 #include "core/framework/program_region.h"
 #endif
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-template <typename T>
-struct Offset;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 
 namespace fbs {
diff --git a/onnxruntime/core/graph/graph_flatbuffers_utils.h b/onnxruntime/core/graph/graph_flatbuffers_utils.h
index f4899ff..b625cbf 100644
--- a/onnxruntime/core/graph/graph_flatbuffers_utils.h
+++ b/onnxruntime/core/graph/graph_flatbuffers_utils.h
@@ -5,6 +5,8 @@
 
 #include <memory>
 
+#include "flatbuffers/flatbuffers.h"
+
 #include "core/common/status.h"
 #include "core/graph/ort_format_load_options.h"
 #include "core/framework/tensor.h"
@@ -18,12 +20,6 @@ class SparseTensorProto;
 #endif  // !defined(DISABLE_SPARSE_TENSORS)
 }  // namespace ONNX_NAMESPACE
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-template <typename T>
-struct Offset;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 
 class Graph;
diff --git a/onnxruntime/core/graph/model.h b/onnxruntime/core/graph/model.h
index 5337211..f705e85 100644
--- a/onnxruntime/core/graph/model.h
+++ b/onnxruntime/core/graph/model.h
@@ -7,6 +7,7 @@
 #include <memory>
 #include <climits>
 #include <string>
+#include "flatbuffers/flatbuffers.h"
 #include "core/common/path.h"
 #include "core/graph/graph_viewer.h"
 #include "core/graph/ort_format_load_options.h"
@@ -15,12 +16,6 @@
 #include "core/graph/function_template.h"
 #endif
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-template <typename T>
-struct Offset;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 
 namespace fbs {
diff --git a/onnxruntime/core/graph/op_identifier_utils.h b/onnxruntime/core/graph/op_identifier_utils.h
index 265364a..8a9351a 100644
--- a/onnxruntime/core/graph/op_identifier_utils.h
+++ b/onnxruntime/core/graph/op_identifier_utils.h
@@ -3,21 +3,14 @@
 
 #pragma once
 
+#include "flatbuffers/flatbuffers.h"
+
 #include "core/graph/op_identifier.h"
 
 #include "core/common/status.h"
 #include "core/graph/graph.h"
 #include "core/graph/onnx_protobuf.h"
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-
-template <typename T>
-struct Offset;
-
-struct String;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 
 namespace fbs::utils {
diff --git a/onnxruntime/core/graph/runtime_optimization_record_container.h b/onnxruntime/core/graph/runtime_optimization_record_container.h
index 5db784f..a28b19e 100644
--- a/onnxruntime/core/graph/runtime_optimization_record_container.h
+++ b/onnxruntime/core/graph/runtime_optimization_record_container.h
@@ -9,17 +9,11 @@
 #include <unordered_map>
 #include <vector>
 
+#include "flatbuffers/flatbuffers.h"
+
 #include "core/common/common.h"
 #include "core/graph/runtime_optimization_record.h"
 
-namespace flatbuffers {
-class FlatBufferBuilder;
-template <typename T>
-struct Offset;
-template <typename T>
-class Vector;
-}  // namespace flatbuffers
-
 namespace onnxruntime {
 
 namespace fbs {
diff --git a/onnxruntime/core/platform/windows/stacktrace.cc b/onnxruntime/core/platform/windows/stacktrace.cc
index cac6f4f..d7d423e 100644
--- a/onnxruntime/core/platform/windows/stacktrace.cc
+++ b/onnxruntime/core/platform/windows/stacktrace.cc
@@ -10,7 +10,6 @@
 #include <stacktrace>
 #endif
 #endif
-#include <stacktrace>
 
 #include "core/common/logging/logging.h"
 #include "core/common/gsl.h"
