diff --git a/cmake/onnxruntime_framework.cmake b/cmake/onnxruntime_framework.cmake
index 15f3105..af6609b 100644
--- a/cmake/onnxruntime_framework.cmake
+++ b/cmake/onnxruntime_framework.cmake
@@ -122,5 +122,5 @@ else()
             ARCHIVE   DESTINATION ${CMAKE_INSTALL_LIBDIR}
             LIBRARY   DESTINATION ${CMAKE_INSTALL_LIBDIR}
             RUNTIME   DESTINATION ${CMAKE_INSTALL_BINDIR}
-            FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR})
+            FRAMEWORK DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
diff --git a/cmake/onnxruntime_webassembly.cmake b/cmake/onnxruntime_webassembly.cmake
index ffe8661..a3e9b4f 100644
--- a/cmake/onnxruntime_webassembly.cmake
+++ b/cmake/onnxruntime_webassembly.cmake
@@ -94,7 +94,7 @@ if (NOT onnxruntime_ENABLE_WEBASSEMBLY_THREADS)
   )
 
   # Override re2 compiler options to remove -pthread
-  set_property(TARGET re2 PROPERTY COMPILE_OPTIONS )
+  # set_property(TARGET re2 PROPERTY COMPILE_OPTIONS )
 endif()
 
 if (NOT onnxruntime_USE_VCPKG)
--- a/cmake/external/onnxruntime_external_deps.cmake
+++ b/cmake/external/onnxruntime_external_deps.cmake
@@ -640,7 +640,7 @@
   # until we figure out how to resolve the packaging pipeline failures
   #
   # if (onnxruntime_USE_VCPKG AND NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
-  if (FALSE)
+  if (onnxruntime_USE_VCPKG AND NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
     # vcpkg does not support Emscripten yet
-     find_package(dawn REQUIRED)
+     find_package(Dawn REQUIRED)
   else()
@@ -828,17 +828,6 @@
     onnxruntime_fetchcontent_makeavailable(dawn)
   endif()
 
-  if (NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
-    if (onnxruntime_BUILD_DAWN_SHARED_LIBRARY)
-      list(APPEND onnxruntime_EXTERNAL_LIBRARIES dawn::webgpu_dawn)
-    else()
-      if (NOT onnxruntime_USE_EXTERNAL_DAWN)
-        list(APPEND onnxruntime_EXTERNAL_LIBRARIES dawn::dawn_native)
-      endif()
-      list(APPEND onnxruntime_EXTERNAL_LIBRARIES dawn::dawn_proc)
-    endif()
-  endif()
-
   if (onnxruntime_ENABLE_PIX_FOR_WEBGPU_EP)
     list(APPEND onnxruntime_EXTERNAL_LIBRARIES webgpu_glfw glfw)
   endif()
diff -ur a/cmake/onnxruntime_providers_webgpu.cmake b/cmake/onnxruntime_providers_webgpu.cmake
--- a/cmake/onnxruntime_providers_webgpu.cmake
+++ b/cmake/onnxruntime_providers_webgpu.cmake
@@ -231,7 +231,7 @@
       if (NOT onnxruntime_USE_EXTERNAL_DAWN)
         target_link_libraries(onnxruntime_providers_webgpu PRIVATE dawn::dawn_native)
       endif()
-      target_link_libraries(onnxruntime_providers_webgpu PRIVATE dawn::dawn_proc)
+      target_link_libraries(onnxruntime_providers_webgpu PRIVATE dawn::webgpu_dawn)
     endif()
 
     if (WIN32 AND onnxruntime_ENABLE_DAWN_BACKEND_D3D12)
