diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0745cde0..802da0e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,7 +146,15 @@
 endif (APPLE)
 
 install(TARGETS fastdeploy_ppocr EXPORT fastdeploy_ppocrConfig)
-install(EXPORT fastdeploy_ppocrConfig DESTINATION share/fastdeploy_ppocr)
+
+install(EXPORT fastdeploy_ppocrConfig
+    CONFIGURATIONS Release
+    DESTINATION share/fastdeploy_ppocr)
+
+install(EXPORT fastdeploy_ppocrConfig
+    CONFIGURATIONS Debug
+    DESTINATION share/fastdeploy_ppocr)
+
 install(
   DIRECTORY ${PROJECT_SOURCE_DIR}/fastdeploy
   DESTINATION ${CMAKE_INSTALL_PREFIX}/include
